### PR TITLE
Add most important CMake options

### PR DESCRIPTION
**Pull request by dimitraka**
Add the agreed most important CMake options in the Manual.

### CODE REVIEW

File: cmake/templates/cmake_variables.rst.in [hunk 1]
```diff
@@ -10,6 +10,61 @@
...
+.. option:: HPX_WITH_TESTS
+   
+   Build tests.
+
+.. option:: HPX_WITH_MALLOC
+
+   Use a custom allocator. Using a custom allocator tuned for multithreaded applications is very 
+   important for the performance of HPX applications. When debugging applications, it's useful to set 
+   this to ``SYSTEM``, as custom allocators can hide some memory-related bugs. Note that setting this to 
+   something other than ``SYSTEM`` requires an external dependency.
```
Comment (msimberg) [inline cmake/templates/cmake_variables.rst.in:35]:
```suggestion
   this to ``system``, as custom allocators can hide some memory-related bugs. Note that setting this to 
   something other than ``system`` requires an external dependency.
```

File: cmake/templates/cmake_variables.rst.in [hunk 2]
```diff
@@ -10,6 +10,61 @@
...
+
+.. option:: HPX_WITH_APEX
+   
+   Enable APEX integration. `APEX <https://uo-oaciss.github.io/apex/quickstarthpx/>`_ can be used to profile |hpx|
+   applications. In particular, it provides information about individual tasks in the |hpx| runtime.
+
+.. option:: HPX_WITH_GENERIC_CONTEXT_COROUTINES
+
+   Enable use of Boost. This is useful for task context switching. It should not be enabled on x86 architechture, 
+   only for architectures such as ARM or Power.
```
Comment (msimberg) [inline cmake/templates/cmake_variables.rst.in:61]:
```suggestion
   Enable Boost.Context for task context switching. It must be enabled for non-x86 architectures such as ARM and Power.
```

File: cmake/templates/cmake_variables.rst.in [hunk 3]
```diff
@@ -10,6 +10,61 @@
    This file has been automatically generated using the CMake tool.
    Do not edit manually.
 
+.. _important_cmake_options:
```
Comment (msimberg) [inline cmake/templates/cmake_variables.rst.in:13]: Would it not be good to put this list somewhere more prominent, like just before this section: https://hpx-docs.stellar-group.org/branches/master/html/manual/building_hpx.html#platform-specific-build-recipes? You could then also include a link from this section to the full list: https://hpx-docs.stellar-group.org/branches/master/html/manual/building_hpx.html#cmake-variables-used-to-configure-hpx.

**Review by msimberg — CHANGES_REQUESTED** [review]
Thanks @dimitraka! A few suggestions in the comments.

File: cmake/templates/cmake_variables.rst.in [hunk 4]
```diff
@@ -10,6 +10,61 @@
...
+   Build examples.
+
+.. option:: HPX_WITH_TESTS
+   
+   Build tests.
+
+.. option:: HPX_WITH_MALLOC
+
+   Use a custom allocator. Using a custom allocator tuned for multithreaded applications is very 
+   important for the performance of HPX applications. When debugging applications, it's useful to set
```
Comment (aurianer) [inline cmake/templates/cmake_variables.rst.in:33]:
```suggestion
   important for the performance of |hpx| applications. When debugging applications, it's useful to set 
```
For the specific font?

File: cmake/templates/cmake_variables.rst.in [hunk 5]
```diff
@@ -10,6 +10,61 @@
...
+.. option:: HPX_WITH_MALLOC
+
+   Use a custom allocator. Using a custom allocator tuned for multithreaded applications is very 
+   important for the performance of HPX applications. When debugging applications, it's useful to set 
+   this to ``SYSTEM``, as custom allocators can hide some memory-related bugs. Note that setting this to 
+   something other than ``SYSTEM`` requires an external dependency.
+
+.. option:: HPX_WITH_CUDA
+
+   Enable support for CUDA. Use ``CMAKE_CUDA_COMPILER`` to set the CUDA compiler. This is a standard CMake variable,
```
Comment (aurianer) [inline cmake/templates/cmake_variables.rst.in:39]:
```suggestion
   Enable support for CUDA. Use ``CMAKE_CUDA_COMPILER`` to set the CUDA compiler. This is a standard |cmake| variable, 
```
?

File: cmake/templates/cmake_variables.rst.in [hunk 6]
```diff
@@ -10,6 +10,61 @@
...
+   applications. In particular, it provides information about individual tasks in the |hpx| runtime.
+
+.. option:: HPX_WITH_GENERIC_CONTEXT_COROUTINES
+
+   Enable use of Boost. This is useful for task context switching. It should not be enabled on x86 architechture, 
+   only for architectures such as ARM or Power.
+
+.. option:: HPX_WITH_MAX_CPU_COUNT
+
+   Set the maximum CPU count supported by HPX. The default value is 64, and should be set to a number at least as
```
Comment (aurianer) [inline cmake/templates/cmake_variables.rst.in:65]:
```suggestion
   Set the maximum CPU count supported by |hpx|. The default value is 64, and should be set to a number at least as
```

**Review by aurianer — DISMISSED** [review]
LGTM, thanks a lot!

File: docs/sphinx/manual/building_hpx.rst [hunk 7]
```diff
@@ -183,6 +183,60 @@ more recent standards can be opted into explicitly. If you want to force |hpx|
 to use a specific C++ standard version, you can use the ``HPX_WITH_CXX_STANDARD``
 CMake variable; e.g. ``HPX_WITH_CXX_STANDARD=20``.
```
Comment (msimberg) [inline docs/sphinx/manual/building_hpx.rst:184]: As discussed in the meeting, let's move this to the list of options below. Just adding this comment as a reminder.

**Review by msimberg — APPROVED** [review]
Looks good. Thanks @dimitraka!

**Comment by msimberg** [comment]
bors merge

**Pull request merged**